Source/Scene/PointPrimitiveCollection.js

```diff
@@ -0,0 +1,976 @@
+        var mode = frameState.mode;
+
+        var pointPrimitives = pointPrimitiveCollection._pointPrimitives;
+        var pointPrimitivesToUpdate = pointPrimitiveCollection._pointPrimitivesToUpdate;
+        var modelMatrix = pointPrimitiveCollection._modelMatrix;
+
+        if (pointPrimitiveCollection._createVertexArray ||
+            pointPrimitiveCollection._mode !== mode ||
+            mode !== SceneMode.SCENE3D &&
+            !Matrix4.equals(modelMatrix, pointPrimitiveCollection.modelMatrix)) {
+
+            pointPrimitiveCollection._mode = mode;
+            Matrix4.clone(pointPrimitiveCollection.modelMatrix, modelMatrix);
+            pointPrimitiveCollection._createVertexArray = true;
+
+            if (mode === SceneMode.SCENE3D || mode === SceneMode.SCENE2D || mode === SceneMode.COLUMBUS_VIEW) {
+                recomputeActualPositions(pointPrimitiveCollection, pointPrimitives, pointPrimitives.length, frameState, modelMatrix, true);
+            }
+        } else if (mode === SceneMode.MORPHING) {
+            recomputeActualPositions(pointPrimitiveCollection, pointPrimitives, pointPrimitives.length, frameState, modelMatrix, true);
+        } else if (mode === SceneMode.SCENE2D || mode === SceneMode.COLUMBUS_VIEW) {
+            recomputeActualPositions(pointPrimitiveCollection, pointPrimitivesToUpdate, pointPrimitiveCollection._pointPrimitivesToUpdateIndex, frameState, modelMatrix, false);
+        }
+    }
+
+    var scratchDrawingBufferDimensions = new Cartesian2();
+    var scratchToCenter = new Cartesian3();
+    var scratchProj = new Cartesian3();
+    function updateBoundingVolume(collection, context, frameState, boundingVolume) {
+        var camera = frameState.camera;
+        var frustum = camera.frustum;
+
+        var toCenter = Cartesian3.subtract(camera.positionWC, boundingVolume.center, scratchToCenter);
+        var proj = Cartesian3.multiplyByScalar(camera.directionWC, Cartesian3.dot(toCenter, camera.directionWC), scratchProj);
+        var distance = Math.max(0.0, Cartesian3.magnitude(proj) - boundingVolume.radius);
+
+        scratchDrawingBufferDimensions.x = context.drawingBufferWidth;
+        scratchDrawingBufferDimensions.y = context.drawingBufferHeight;
+        var pixelSize = frustum.getPixelSize(scratchDrawingBufferDimensions, distance);
+        var pixelScale = Math.max(pixelSize.x, pixelSize.y);
+
+        var size = pixelScale * collection._maxPixelSize;
+        boundingVolume.radius += size;
+    }
+
+    /**
+     * @private
+     */
+    PointPrimitiveCollection.prototype.update = function(context, frameState, commandList) {
+        removePointPrimitives(this);
+
+        this._maxTotalPointSize = context.maximumAliasedPointSize;
+
+        var pointPrimitives = this._pointPrimitives;
+        var pointPrimitivesLength = pointPrimitives.length;
+
+        updateMode(this, frameState);
+
+        pointPrimitives = this._pointPrimitives;
```

Comment:
This and the line below are redundant with the two lines above.

Did you step through this `update` function for a few common cases to make sure that it is not doing anything it doesn't need to?  For example:
* When all points are new
* When all points didn't change
* When all points changed
* When one point changed